docs/developers/release-checklist.md

```diff
@@ -0,0 +1,30 @@
+## `ddev` Release Checklist 
+
+- [ ] [Create a release](https://github.com/drud/ddev/releases/new) using the github web UI. This will initiate a CircleCI tag build.
+- [ ] Check out the ddev source with `git fetch upstream && git checkout <release_tag>`
+- [ ] In each directory if containers/ do a `make push`.
+- [ ] Add the commit list to the release page
+- [ ] Download artifacts from CircleCI and upload them (except for the Windows installer and its sha256.txt) to the release page
+- [ ] Update the `ddev` [Homebrew formula](https://github.com/drud/homebrew-ddev) with the MacOS `.tar.gz` and SHA checksum
+- [ ] Test `brew upgrade ddev` and make sure ddev is the right version and behaves well.
+- [ ] Sign the Windows installer and create a new sha256.txt according to steps below.
```

Comment:
Small nit, but getting consistent periods at the end of each sentence would be great!